### PR TITLE
Add and implement new front end money input parser

### LINKED ISSUE #3: Need new front end money input parser
<h3>The current process to input amounts of money is little clunky and could be improved.</h3>

---

<h4>Problem:</h4>

If a decimal is needed, first the entire number without the decimal must be typed, and then the decimal must be placed, like so:

```
12345  -->   123.45
```

Also, if a negative number must be input, a similar pattern must be followed:

```
12345  -->  -12345
```

---

<h4>Solution:</h4>

What I propose is having the cursor be fixed at the end of the number, and have new numbers push the old numbers one decimal place to the left, like so:

```
0.00  -->  0.01  -->  0.12  -->  1.23  -->  12.34  -->  123.45
```

In this example, the cursor position never changes: it is fixed on the right side of the number. 
Every new number that is input is placed at the end of the current number and displaces the previous numbers one place to the left.

### PR DESCRIPTION
**Pull request by redraze**
## tl;dr
- Resolves #3
- Enables more intuitive dollar amount entry in form inputs found in endpoints like /balances and /payments
- Handles decimal place changes and sign changes

---

### Before, 
dollar amounts with a decimal place needed to be entered the following way:
```
0 --> 1 --> 12 --> 123 --> 12.3
```
where the decimal place had to placed after a portion or the entire number was input.

### Now,
dollar amounts and dollar amount changes are handled in a more intuitive way. Like so:
```
0.00 --> 0.01 --> 0.12 --> 1.23 --> 12.34 --> ...
```
where the cursor is placed by default at the beginning of the number, and any new input 'pushes' the previous amount over by one decimal place. 

Middle numbers also properly displace already existing numbers, like so:
```
123.45 --> 1263.45
```
or, if entering a new decimal place:
```
123.45 --> 1234.65
```
where upon entering the number '6' in the tenth's place, the number '4' gets pushed to the one's place to make room for the new tenth's place number.

---

### Negative number input is also handled more intuitively.

Now, whenever the negative sign key is detected as an input, the number simply has it's sign switched.
```
123.45 ---> -123.45
```
The placement of the negative sign doesn't make a difference.

### CODE REVIEW

File: app/components/balances/BalanceCorrection.tsx [hunk 1]
```diff
@@ -104,64 +105,57 @@ export default function BalanceCorrection(
...
+                            setBalanceChange(num);
+                            setNewBalance(currentBalance + num);
+                        } }
+                        onChange={ (e) => {
+                            const num = formatVal(e.currentTarget.value);
+                            if (isNaN(num)) { return };
+                            
+                            const diff = currentBalance + num;
+                            setBalanceChange(num);
+                            setNewBalance(diff);
```
Comment (redraze) [inline app/components/balances/BalanceCorrection.tsx:130]: could simplify by writing 
```
setNewBalance(currentBalance + num)
```
instead of having line 128 separate

File: app/components/balances/BalanceCorrection.tsx [hunk 2]
```diff
@@ -104,64 +105,57 @@ export default function BalanceCorrection(
...
+                            setBalanceChange(num - currentBalance);
+                            setNewBalance(num);
+                        } }
+                        onChange={ e => {
+                            const num = formatVal(e.currentTarget.value);
+                            if (isNaN(num)) { return };
+                    
+                            const diff = num - currentBalance;
+                            setBalanceChange(diff);
+                            setNewBalance(num);
```
Comment (redraze) [inline app/components/balances/BalanceCorrection.tsx:153]: again, could simplify code here

**Review by redraze — COMMENTED** [review]
just a few small changes, otherwise looks good!

File: app/components/balances/BalanceCorrection.tsx [hunk 3]
```diff
@@ -104,64 +105,55 @@ export default function BalanceCorrection(
...
+                            const num = Number(e.currentTarget.value) * -1;
+                            setBalanceChange(num);
+                            setNewBalance(currentBalance + num);
+                        } }
+                        onChange={ (e) => {
+                            const num = formatVal(e.currentTarget.value);
+                            if (isNaN(num)) { return };
+                            
+                            setBalanceChange(num);
+                            setNewBalance(currentBalance + num);
```
Comment (redraze) [inline app/components/balances/BalanceCorrection.tsx:129]: looks good!

**Review by redraze — COMMENTED** [review]
just a few last-minute changes, done!
